File: DummyClient/DummyClient/ClientSecurity.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ServiceStack;
using System.IO;

namespace DummyClient
{
    public class ApiCustomHttpHeaders
    {
        public static string UserId = "VEWS-USERID";
        public static string DeviceId = "VEWS-DEVICEID";
        public static string Signature = "VEWS-SIGNATURE";
        public static string SignatureMethod = "VEWS-SIGN_METHOD";
        public static string VEWSVersion = "VEWS-VERSION";
        public static string ReqId = "VEWS-REQID";
        public static string TimeStamp = "VEWS-RESTIME";
    }

    public static class ClientSecurity
    {
        public static void AddRequestHeaders(System.Net.HttpWebRequest request)
        {
            request.ContentType = MimeTypes.Json;

            //Get information to make request   
            //UserId                  
            string userIdStr = "1";
            //DeviceId
            string deviceIdStr = "10";
            //Signature
            //var token = ApiSignature.CreateToken(request, reqDto == null ? string.Empty : reqDto.SerializeToString(), _secret);
            string signatureStr = "feagiserg";
            //SignatureMethod
            string signatureMethodStr = "HMAC_SHA256";
            //VEWS version
            string vewsVersionStr = "V1R1";
            //TimeStamp
            //System.DateTime initDate = new DateTime(1969, 12, 31, 23, 54, 59);
            System.DateTime initDate = new DateTime(1970, 1, 1, 0, 0, 0);
            TimeSpan tsUtcTime = new TimeSpan(System.DateTime.UtcNow.Ticks - initDate.Ticks);
            string timeStampStr = ((ulong)tsUtcTime.TotalMilliseconds).ToString();
            //RequestId
            string requestIdStr = userIdStr + deviceIdStr + timeStampStr;

            //Add custom fields to header
            request.Headers.Add(ApiCustomHttpHeaders.UserId, userIdStr);            
            request.Headers.Add(ApiCustomHttpHeaders.DeviceId, deviceIdStr);
            request.Headers.Add(ApiCustomHttpHeaders.Signature, signatureStr);
            request.Headers.Add(ApiCustomHttpHeaders.SignatureMethod, signatureMethodStr);
            request.Headers.Add(ApiCustomHttpHeaders.VEWSVersion, vewsVersionStr);            
            request.Headers.Add(ApiCustomHttpHeaders.TimeStamp, timeStampStr);
            request.Headers.Add(ApiCustomHttpHeaders.ReqId, requestIdStr);            
            request.SendChunked = false;
        }

        public static void ValidateResponseHeaders(System.Net.HttpWebResponse response)
        {
            string serverTimestamp = response.Headers[ApiCustomHttpHeaders.TimeStamp];
        }
    }
}

File: DummyClient-Java/DummyServiceStack/src/dummyservicestack/ClientSecurity.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package dummyservicestack;

import com.google.gson.Gson;
import com.google.gson.JsonElement;
import com.google.gson.JsonObject;
import static com.oracle.jrockit.jfr.DataType.UTF8;
import java.io.IOException;
import java.io.InputStream;
import java.net.HttpURLConnection;
import java.util.Map;
import net.servicestack.client.HttpHeaders;
import net.servicestack.client.MimeTypes;
import net.servicestack.client.Utils;
import net.servicestack.client.WebServiceException;

/**
 *
 * @author Cesar
 */
public class ClientSecurity {
   public static void AddRequestHeaders(HttpURLConnection request)
   {
     
            //Get information to make request   
            //UserId                  
            String userIdStr = "1";
            //DeviceId
            String deviceIdStr = "10";
            //Signature
            String signatureStr = "feagiserg";
            //SignatureMethod
            String signatureMethodStr = "HMAC_SHA256";
            //VEWS version
            String vewsVersionStr = "V1R0";
            //TimeStamp
            String timeStampStr = Long.toString(System.currentTimeMillis());
            //RequestId
            String requestIdStr = userIdStr + deviceIdStr + timeStampStr;
            
            request.addRequestProperty(ApiCustomHttpHeaders.UserId, userIdStr);
            request.addRequestProperty(ApiCustomHttpHeaders.DeviceId, deviceIdStr);
            request.addRequestProperty(ApiCustomHttpHeaders.Signature, signatureStr);
            request.addRequestProperty(ApiCustomHttpHeaders.SignatureMethod, signatureMethodStr);
            request.addRequestProperty(ApiCustomHttpHeaders.VEWSVersion, vewsVersionStr);
            request.addRequestProperty(ApiCustomHttpHeaders.TimeStamp, timeStampStr);
            request.addRequestProperty(ApiCustomHttpHeaders.ReqId, requestIdStr);
            request.setUseCaches(false);
            request.setDoInput(true);
            request.setDoOutput(true);
            
   }

   public static void ValidateResponseHeaders(HttpURLConnection response)
        {
            String serverTimestamp = response.getHeaderField(ApiCustomHttpHeaders.TimeStamp);
        }
}
